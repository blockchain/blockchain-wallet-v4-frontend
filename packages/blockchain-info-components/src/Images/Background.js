import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import Images from './Images'

const BaseBackground = styled.div`
  background: url(${props => props.url});
  background-size: cover;
  background-position: 70% 100%;
  width: ${props => props.width};
  height: ${props => props.height};
`

const Background = props => {
  const { name, ...rest } = props
  const file = Images[name]

  return <BaseBackground url={file} {...rest} />
}

Background.defaultProps = {
  width: 'auto',
  height: 'auto'
}

Background.propTypes = {
  name: PropTypes.string.isRequired,
  width: PropTypes.string,
  height: PropTypes.string
}

export default Background
