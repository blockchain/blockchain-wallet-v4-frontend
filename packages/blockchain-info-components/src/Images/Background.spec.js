import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Background from './Background'

describe('Background component', () => {
  it('default renders correctly', () => {
    const component = shallow(<Background name='testing' width='10px' height='10px' />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
