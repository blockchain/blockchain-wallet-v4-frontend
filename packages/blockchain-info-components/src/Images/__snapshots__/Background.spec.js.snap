// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background component default renders correctly 1`] = `
<Background__BaseBackground
  height="10px"
  width="10px"
/>
`;
